test/utils/shoots/update/highavailability/update.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package highavailability
+
+import (
+	"context"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/test/framework"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// UpdateAndVerify runs the HA control-plane update tests for an existing shoot cluster.
+func UpdateAndVerify(
+	ctx context.Context,
+	f *framework.ShootFramework,
+	failureToleranceType gardencorev1beta1.FailureToleranceType,
+
+) {
+	By("Update Shoot control plane to HA with failure tolerance type " + string(failureToleranceType))
+	Expect(f.UpdateShoot(ctx, func(shoot *gardencorev1beta1.Shoot) error {
+		shoot.Spec.ControlPlane = &gardencorev1beta1.ControlPlane{
+			HighAvailability: &gardencorev1beta1.HighAvailability{
+				FailureTolerance: gardencorev1beta1.FailureTolerance{
+					Type: failureToleranceType,
+				},
+			},
+		}
+		return nil
+	})).To(Succeed())
+
+	By("Verify Shoot's control plane components")
+	verifyTSC(ctx, f.SeedClient, f.Shoot, f.ShootSeedNamespace())
```

Comment:
Prevent using "cryptic" abbreviations, i.e. call it `TopologySpreadConstraint`s everywhere to make it easier readable